File: src/main/java/com/avant/auth/AuthToken.java
package com.avant.auth;

import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;

import java.util.UUID;

@Data
@NoArgsConstructor
public class AuthToken {

	@Id
	private String token;
	@DBRef
	private User user;
	private long timeout;

	public AuthToken(User user) {
		this.user = user;
		this.token = UUID.randomUUID().toString();
		this.timeout = System.currentTimeMillis() + 7_200_000L;
	}
}

File: src/main/java/com/avant/auth/AuthToken.kt
package com.avant.auth

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.DBRef

import java.util.UUID

class AuthToken(@DBRef private val user: User) {
	
	@Id
	private val token: String = UUID.randomUUID().toString()
	private val timeout: Long = System.currentTimeMillis() + 7_200_000L
	
}